### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by Progress2002**
## The following was implemented :

- Added Redux Toolkit(npm install react-redux @reduxjs/toolkit).
- Wrote book's actions and reducer (in /src/redux/books/books.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).
- Wrote categories actions and reducer (in /src/redux/categories/categories.js) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).
- Configured the Redux Store (/src/redux/configureStore.js):
- Used git-flow

### CODE REVIEW

File: src/Components/redux/books/books.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
+import types from '../types/types';
+
+// initial state
+const initialState = {
```
Comment (bobb-Rob) [inline src/Components/redux/books/books.js]: - [ ] According to the project requirement, your file structure should be something like this 👇🏾 
image

Please fix this.

File: src/Components/redux/configuration.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+import { combineReducers, configureStore } from '@reduxjs/toolkit';
+import booksReducer from './books/books';
+import categoriesReducer from './categories/categories';
+
+// combine all reducers together
+
+const reducer = combineReducers({
+  book: booksReducer,
```
Comment (bobb-Rob) [inline src/Components/redux/configuration.js]: - [ ] [OPTIONAL] Line 8 `book` should be plural as in "books" instead.

**Review by bobb-Rob — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️
Hi @Progress2002 , 

Good job so far! 👏👏👏 
There are some issues that you still need to work on to go to the next project but you are almost there!  😃 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Highlights
✅ No Linter error
✅ Correct use of Gitflow
✅ Very descriptive Pull Request(PR)
✅ Neat Commit messages

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/Components/redux/configuration.js (hunk 2 above)
Comment (Progress2002) [reply src/Components/redux/configuration.js]: Thanks for the suggestion, file has been updated. Thank you.

**Review by DammyShittu — APPROVED** [review]
Hi @Progress2002,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**